### PR TITLE
Fix create_data_dict for work with list

### PR DESCRIPTION
**Pull request by Leits**
[Review on Reviewable](https://reviewable.io/reviews/openprocurement/robot_tests/56)

### CODE REVIEW

File: op_robot_tests/tests_files/service_keywords.py [hunk 1]
```diff
@@ -185,7 +185,11 @@ def merge_dicts(left, right):
 def create_data_dict(path_to_key=None, value=None):
     data_dict = munchify({'data': {}})
     if isinstance(path_to_key, basestring) and isinstance(value, basestring):
-        data_dict = set_to_object(data_dict, path_to_key, value)
+        if '[0]' in path_to_key:
```
Comment (selurvedu) [inline op_robot_tests/tests_files/service_keywords.py]: А якщо в нас в шляху `[1]` або інше число? А якщо `-1`?

File: op_robot_tests/tests_files/service_keywords.py [hunk 2]
```diff
@@ -185,7 +186,12 @@ def merge_dicts(left, right):
 def create_data_dict(path_to_key=None, value=None):
     data_dict = munchify({'data': {}})
     if isinstance(path_to_key, basestring) and isinstance(value, basestring):
-        data_dict = set_to_object(data_dict, path_to_key, value)
+        list_len = re.search('\d+', path_to_key).group(0)
+        if list_len:
+            path_to_key = path_to_key.replace('[', '').replace('].', '').split(list_len)
+            data_dict = set_to_object(data_dict, path_to_key[0] + list_len, set_to_object({}, path_to_key[1], value))
+        else:
+            data_dict = set_to_object(data_dict, path_to_key, value)
```
Comment (selurvedu) [inline op_robot_tests/tests_files/service_keywords.py]: Вже краще, але давай спробуй щось з `dpath.util.new()` в `set_to_object`. А раптом `dpath` вміє сам створювати лісти?

Comment (selurvedu) [inline op_robot_tests/tests_files/service_keywords.py]: І поміняй в set_to_object()
```python
xpathset(obj, attribute.replace('.', '/'), value)
```
на 
```python
xpathset(obj, attribute, value, separator='.')
```

File: op_robot_tests/tests_files/service_keywords.py [hunk 3]
```diff
@@ -182,10 +183,19 @@ def merge_dicts(left, right):
     return new
 
 
-def create_data_dict(path_to_key=None, value=None):
+def create_data_dict(path_to_value=None, value=None):
     data_dict = munchify({'data': {}})
-    if isinstance(path_to_key, basestring) and isinstance(value, basestring):
-        data_dict = set_to_object(data_dict, path_to_key, value)
+    if isinstance(path_to_value, basestring) and value:
+        list_items = re.search('\d+', path_to_value)
```
Comment (selurvedu) [inline op_robot_tests/tests_files/service_keywords.py]: Шо-то тут нє то:
```python
>>> x = {}
>>> set_to_object(x, 'foo.bar[0]', True)
{'foo': {'bar[0]': True}}
>>> set_to_object(x, 'foo.bar', [])
{'foo': {'bar': [], 'bar[0]': True}}
>>> set_to_object(x, 'foo.bar[0]', False)
{'foo': {'bar': [], 'bar[0]': False}}
>>> set_to_object(x, 'foo.bar.0', True)
{'foo': {'bar': [True], 'bar[0]': True}}
>>> 
```

**Comment by myroslav** [comment]
Reviewed 1 of 1 files at r1.
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/openprocurement/robot_tests/56)*

**Comment by myroslav** [comment]
Reviewed 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/openprocurement/robot_tests/56)*

**Pull request merged**